libraries/AP_Scripting/lua_scripts.cpp

```diff
@@ -0,0 +1,336 @@
+                   break;
+                }
+            default:
+                {
+                    gcs().send_text(MAV_SEVERITY_CRITICAL, "Lua: %s returned bad result count (%d)", script->name, returned);
+                    remove_script(L, script);
+                    // pop all the results we got that we didn't expect
+                    lua_pop(L, returned);
+                    break;
+                 }
+         }
+     }
+}
+
+void lua_scripts::remove_script(lua_State *L, script_info *script) {
+    // ensure that the script isn't in the loaded list for any reason
+    if (scripts == nullptr) {
+        // nothing to do, already not in the list
+    } else if (scripts == script) {
+        scripts = script->next;
+    } else {
+        script_info * previous = scripts;
+        while (previous->next != nullptr) {
+            if (previous->next == script) {
+                previous->next = script->next;
+                break;
+            }
+            previous = previous->next;
+        }
+    }
+
+    luaL_unref(L, LUA_REGISTRYINDEX, script->lua_ref);
+    luaM_free(L, script->name);
+    luaM_free(L, script);
+}
+
+void lua_scripts::reschedule_script(script_info *script) {
+    if (script == nullptr) {
+       #if defined(AP_SCRIPTING_CHECKS) && AP_SCRIPTING_CHECKS >= 1
+       AP_HAL::panic("Lua: Attempted to schedule a null pointer");
+       #endif // defined(AP_SCRIPTING_CHECKS) && AP_SCRIPTING_CHECKS >= 1
+       return;
+    }
+
+    script->next = nullptr;
+    if (scripts == nullptr) {
+        scripts = script;
+        return;
+    }
+
+    uint64_t next_run_ms = script->next_run_ms;
+
+    if (scripts->next_run_ms > next_run_ms) {
+        script->next = scripts;
+        scripts = script;
+        return;
+    }
+
+    script_info *previous = scripts;
+    while (previous->next != nullptr) {
+        if (previous->next->next_run_ms > next_run_ms) {
+            script->next = previous->next;
+            previous->next = script;
+            return;
+        }
+        previous = previous->next;
+    }
+
+    previous->next = script;
+}
+
+void lua_scripts::run(void) {
+    lua_State *L = luaL_newstate();
+    luaL_openlibs(L);
+    load_lua_bindings(L);
+
+    // load the sandbox creation function
+    uint32_t sandbox_size;
+    char *sandbox_data = (char *)AP_ROMFS::find_decompress("sandbox.lua", sandbox_size);
+    if (sandbox_data == nullptr) {
+        gcs().send_text(MAV_SEVERITY_CRITICAL, "Scripting: Could not find sandbox");
+        return;
+    }
+
+    if (luaL_dostring(L, sandbox_data)) {
+        gcs().send_text(MAV_SEVERITY_CRITICAL, "Scripting: Loading sandbox: %s", lua_tostring(L, -1));
+        return;
+    }
+    free(sandbox_data);
+
+    // Scan the filesystem in an appropriate manner and autostart scripts
+    load_all_scripts_in_dir(L, SCRIPTING_DIRECTORY);
+
+    while (true) {
+        if (scripts != nullptr) {
+          #if defined(AP_SCRIPTING_CHECKS) && AP_SCRIPTING_CHECKS >= 1
```

Comment:
Nitpicking the hell out of pre-processor indentation! Left-align this! 🤣

Out of jokes, at least be consistent on it 😉